modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStore.java

```diff
@@ -320,10 +320,10 @@ String toSingleLineWKT(CoordinateReferenceSystem crs) {
      * Attempt to create a DbaseFileHeader for the FeatureType. Note, we cannot set the number of
      * records until the write has completed.
      */
-    protected static DbaseFileHeader createDbaseHeader(SimpleFeatureType featureType)
-            throws IOException, DbaseFileException {
+    protected static DbaseFileHeader createDbaseHeader(
+            SimpleFeatureType featureType, Charset charset) throws IOException, DbaseFileException {
```

Comment:
We tend not to backport API changes, I understand that this a protected constructor so the change is "internal".